invokeai/app/api/routers/models.py

```diff
@@ -90,23 +49,28 @@ async def list_models(
     return models
 
 @models_router.post(
-    "/",
+    "/{base_model}/{model_type}/{model_name}",
     operation_id="update_model",
     responses={200: {"status": "success"}},
 )
 async def update_model(
-    model_request: CreateModelRequest
+        base_model: BaseModelType = Path(default='sd-1', description="Base model"),
+        model_type: ModelType = Path(default='main', description="The type of model"),
+        model_name: str = Path(default=None, description="model name"),
+        info: Union[tuple(MODEL_CONFIGS)]  = Body(description="Model configuration"),
 ) -> CreateModelResponse:
     """ Add Model """
-    model_request_info = model_request.info
-    info_dict = model_request_info.dict()
-    model_response = CreateModelResponse(name=model_request.name, info=model_request.info, status="success")
-
     ApiDependencies.invoker.services.model_manager.add_model(
-        model_name=model_request.name,
-        model_attributes=info_dict,
+        model_name=model_name,
+        base_model=base_model,
+        model_type=model_type,
+        model_attributes=info.dict(),
         clobber=True,
     )
+    model_response = CreateModelResponse(
+        model_name = model_name,
+        info = info,
+        status="success")
```

Comment:
Doesn't the `info` include the model name? The status is implied by a 200-class response, so it should just return the updated model

I believe the convention would be to mutate the model, then retrieve it again and return the retrieved model instead.